test/modules/TestResultCoordinator/Services/TestResultReportingService.cs

```diff
@@ -22,25 +22,39 @@ class TestResultReportingService : IHostedService, IDisposable
         readonly TimeSpan delayBeforeWork;
         readonly ITestOperationResultStorage storage;
         readonly TestResultReportingServiceSettings serviceSpecificSettings;
+        readonly TimeSpan sendReportFrequency;
         Timer timer;
 
         public TestResultReportingService(ITestOperationResultStorage storage)
         {
             this.storage = Preconditions.CheckNotNull(storage, nameof(storage));
-            this.delayBeforeWork = Settings.Current.TestStartDelay + Settings.Current.TestDuration + Settings.Current.DurationBeforeVerification;
+            if (TestMode.Connectivity.Equals(Settings.Current.TestMode))
+            {
+                ConnectivitySpecificSettings connectivitySettings =
+                    Settings.Current.ConnectivitySpecificSettings.Expect(() => new ArgumentException("ConnectivitySpecificSettings must be supplied."));
+                this.delayBeforeWork = Settings.Current.TestStartDelay +
+                    connectivitySettings.TestDuration +
+                    connectivitySettings.TestVerificationDelay;
+            }
+            else
+            {
+                // In long haul mode, wait 1 report frequency before starting
+                this.delayBeforeWork = Settings.Current.SendReportFrequency;
+            }
+
             this.serviceSpecificSettings = Settings.Current.TestResultReportingServiceSettings.Expect(() => new ArgumentException("TestResultReportingServiceSettings must be supplied."));
+            this.sendReportFrequency = Settings.Current.SendReportFrequency;
         }
 
         public Task StartAsync(CancellationToken ct)
         {
             this.logger.LogInformation("Test Result Reporting Service running.");
 
-            // Specify -1 ms to disable periodic run
             this.timer = new Timer(
                 this.DoWorkAsync,
                 null,
                 this.delayBeforeWork,
-                TimeSpan.FromMilliseconds(-1));
+                this.sendReportFrequency);
```

Comment:
The sendReportFrequency for the connectivity tests shouldn't be relevant because the test is over after delayBeforeWork elapses.